packages/react-instantsearch/src/connectors/connectScrollTo.js

```diff
@@ -32,6 +32,27 @@ export default createConnector({
       null,
       currentRefinement => currentRefinement
     );
-    return { value };
+
+    /*
+      We need to keep track of the search state to know if there's a change in the app
+      that was not triggered by the props.scrollOn (id) or the Configure widget. 
+      This is useful when using ScrollTo in combination of Pagination. As pagination
+      can be change by every widget, we want to scroll only if it cames from the pagination
+      widget itself. We also remove the configure key to do this comparaison because for 
+      now configure values are not present in the search state before a first refinements and will false 
+      the results. 
+      See: https://github.com/algolia/react-instantsearch/issues/164
+    */
+    const cleanedSearchState = cleanUpValue(
+      cleanUpValue(searchState, this.context, 'configure'),
+      this.context,
+      id
+    );
+
+    const hasNotChanged = isEqual(this._prevSearchState, cleanedSearchState);
```

Comment:
Can you confirm that:
- this deepEqual comparison is done at every refinement (not at every scroll event)
- what we are doing deep equal on is react-instantsearch state (not helper state)